module/entities/helper.js

```diff
@@ -926,106 +928,75 @@ module.exports = class EntitiesHelper {
-                    { "metaInformation.addressLine2": new RegExp(searchText, 'i') }
-                ];
-
-                let entityDocuments = await database.models.entities.aggregate([
-                    queryObject,
-                    {
-                        $project: {
-                            name: "$metaInformation.name",
-                            externalId: "$metaInformation.externalId",
-                            addressLine1: "$metaInformation.addressLine1",
-                            addressLine2: "$metaInformation.addressLine2",
-                            districtName: "$metaInformation.districtName"
-                        }
-                    },
-                    {
-                        $facet: {
-                            "totalCount": [
-                                { "$count": "count" }
-                            ],
-                            "data": [
-                                { $skip: pageSize * (pageNo - 1) },
-                                { $limit: pageSize }
-                            ],
-                        }
-                    }, {
-                        $project: {
-                            "data": 1,
-                            "count": {
-                                $arrayElemAt: ["$totalCount.count", 0]
-                            }
-                        }
-                    }
-                ]);
-
+                let totalcount = entitiesData.data.count;
+                let immediateEntities = entitiesData.data.response;
+                
+                entityDocuments.push({
+                    data : immediateEntities,
+                    count : totalcount
+                })
                 return resolve(entityDocuments);
-
             } catch (error) {
                 return reject(error);
             }
         })
     }
 
-    /**
+
+/**
    * validate entities.
    * @method 
    * @name validateEntities
-   * @param {String} entityTypeId - Entity type id.
+   * @param {String} entityType - Entity type.
    * @param {Array} entityIds - Array of entity ids.
    */
-
-    static validateEntities(entityIds, entityTypeId) {
-        return new Promise(async (resolve, reject) => {
-            try {
-                let ids = []
-
-                let entitiesDocuments = await database.models.entities.find(
-                    {
-                        _id: { $in: gen.utils.arrayIdsTobjectIds(entityIds) },
-                        entityTypeId: entityTypeId
-                    },
-                    {
-                        _id: 1
-                    }
-                ).lean();
-
-                if (entitiesDocuments.length > 0) {
-                    ids = entitiesDocuments.map(entityId => entityId._id);
-                }
-
-                return resolve({
-                    entityIds: ids
-                });
+    
+ static validateEntities(entityIds, entityType) {
```

Comment:
@VISHNUDAS-tunerlabs check validateEntities is calling any other places and check entityType passing

Reply:
checked. entity id is passing